apis/iam/v1beta1/rolepolicy_types.go

```diff
@@ -0,0 +1,87 @@
+/*
+Copyright 2023 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta1
+
+import (
+	xpv1 "github.com/crossplane/crossplane-runtime/apis/common/v1"
+	extv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type RolePolicyDocument extv1.JSON
+
+// RolePolicyParameters define the desired state of an AWS IAM Role Inline Policy.
+type RolePolicyParameters struct {
+
+	// The JSON policy document that is the content for the policy.
+	Document RolePolicyDocument `json:"document"`
```

Comment:
I think we have to use plain `extv1.JSON` because its custom JSON marshalling methods are lost when creating a new type.